Readme.md

```diff
@@ -1,5 +1,137 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("../src/tests/bnf/fender.bnf");
+```
+
+Then you need to setup a matcher to be able to parse the bnf
```

Comment:
I'm not sure what you mean by "matcher" here. If you meant the match statement, that isn't needed, it's just how the result was handled in this case. It's probably better to just point out that they should use `flux_bnf::bnf::parse` to parse their BNF file. In the readme example code, you can use `.unwrap()` instead of the bulky match statement to get the point across.